partner_industry_parent/models/res_partner.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import SUPERUSER_ID, _, api, models
+
+
+class Partner(models.Model):
+    _inherit = "res.partner"
+
+    @api.model
+    def _search(
```

Comment:
This is just a minor point, I know you didn't write yourself, and purely opinion. I much prefer these overrides where only args is amended to use the idiom 

```python
def _search(args, **kwargs):
   # modify args code
  return super()._search(args, **kwargs)
```
It is loads shorter, more resistant to api changes, and cleaner for people to read as they know only args is being affected.

Then super()